### PR TITLE
Rapiers

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Adds in 2 new rapiers for science and medical
Science rapiers on hit of a dead mob or when killing a mob automatically dusts said mob and gains 50 SI points for research
Medial rapiers work like a sword + needle for injections

Makes it so items can off-set click cooldown

### CODE REVIEW

File: code/game/objects/items/weapons/tools/simple_weapons.dm [hunk 1]
```diff
@@ -426,6 +426,100 @@
 	armor_penetration = ARMOR_PEN_SHALLOW
 	price_tag = 325
 
+/obj/item/tool/sword/saber/deconstuctive_rapier
+	name = "Cinq Deconstructive Rapier"
+	desc = "A long, thin bladed sword with a weaponized destructive analyzer integrated into the blade. At the moment of its victim's death it is able to deconstruct them and wirelessly transmit data to interinal research database."
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/weapons/tools/simple_weapons.dm:431]:
```suggestion
	desc = "A long, thin bladed sword with a weaponized destructive analyzer integrated into the blade. At the moment of its victim's death it is able to deconstruct them and wirelessly transmit data to internal research database."
```

**Comment by benj8560** [comment]
I touched simple_weapons.dm in #4967 as a warning

**Comment by cdb-is-not-good** [comment]
The sprites could probably use some work. We've denied sprites for less

File: code/game/objects/items/weapons/tools/simple_weapons.dm [hunk 2]
```diff
@@ -426,6 +426,100 @@
...
+
+/obj/item/tool/sword/saber/deconstuctive_rapier/resolve_attackby(atom/target, mob/user)
+	.=..()
+	//Little icky but it works
+	if(isliving(target))
+		var/mob/M = target
+		if(!ishuman(M))
+			if(!issilicon(M))
+				if(is_dead(M))
+					user.visible_message("[user] drives [src.name] into [M.name]'s body, decontructing it!", "You drive the [src.name] into [M.name], exstracting value data")
```
Comment (benj8560) [inline code/game/objects/items/weapons/tools/simple_weapons.dm:469]:
```suggestion
					user.visible_message("[user] drives [src.name] into [M.name]'s body, deconstructing it!", "You drive the [src.name] into [M.name], extracting research data")
```

Comment (benj8560) [reply code/game/objects/items/weapons/tools/simple_weapons.dm:469]: not seeing any other trilglish. think that's it.

File: code/game/objects/structures/crates_lockers/closets/secure/medical.dm [hunk 3]
```diff
@@ -190,6 +190,7 @@
 	new /obj/item/circuitboard/sleeper/hyper(src)
 	new /obj/item/oddity/chem_book(src)
 	new /obj/item/storage/hcases/med(src)
+	new /obj/item/tool/sword/saber/deconstuctive_rapier(src)
```
Comment (benj8560) [inline code/game/objects/structures/crates_lockers/closets/secure/medical.dm:193]:
```suggestion
	new /obj/item/tool/sword/saber/injection_rapier(src)
```

Under the impression CBO is suppose to get the medical one not the research one.

**Comment by cdb-is-not-good** [comment]
I'm a *lil* iffy on the sprites, i'd like someone else to chime in here but I dont know that the sprites are up to spec.

**Comment by benj8560** [comment]
> I'm a _lil_ iffy on the sprites still, i'd like someone else to chime in here but I dont know that the sprites are up to spec.

I don't think there so bad to warrant rejecting it. Medical one looks kinda nice to me. Tho I do think they should get on body sprites they seem to just be using the ones from its parent item that don't seem to really fit its inhand sprite.  If I like these enough in game? I might go out of my way to just have pretty sprites paid for if Trilby allows it.

**Comment by AysheDaArtist** [comment]
FINALLY we're getting a cool sword like the WO
BLESS!

On the subject of the sprites themselves, the CBO one looks more like a foil sword ( no tip ) but the CRO one looks fine.

In-hand and on body sprites would be nice, but also, you'll need a scabbard, and perhaps a custom scabbard might be called for here

Could touch these up a bit, just curious what "up to spec." means?  Do we have a design document, do we always shade bottom up?  This is news to me.  CRO sword looks fine, great even, it's cool as hell, the CBO one with no tip is a bit weird, so I'll submit my own take on it here tonight!

**Comment by cdb-is-not-good** [comment]
We don't have a desiign doc in that sense, but to clarify what i mean by that is that the sprites should be comparable in aesthetics to other swords in terms of design. I realize that a rapier is necessarily going to have a bit less room to work with due to it being thinner than a saber(which is a distiinction that can become hard to make with a 32x32 sprite). I'd appreciate if you gave it a tweak towards this end

**Comment by AysheDaArtist** [comment]
RAPIERS_SHOW

**Pull request merged**